doc/manual/interfaces.rst

```diff
@@ -10,20 +10,35 @@ A lot of the power and extensibility in Julia comes from a collection of informa
 :func:`start(iter) <start>`                                                 Returns the initial iteration state
 :func:`next(iter, state) <next>`                                            Returns the current item and the next state
 :func:`done(iter, state) <done>`                                            Tests if there are any items remaining
 **Important optional methods**                     **Default definition**   **Brief description**
+:func:`iteratorsize(IterType) <iteratorsize>`      ``HasLength()``          One of `HasLength()`, `HasShape()`, `IsInfinite()`, or `SizeUnknown()` as appropriate
+:func:`iteratoreltype(IterType) <iteratoreltype>`  ``HasEltype()``          Either `EltypeUnknown()` or `HasEltype()` as appropriate
 :func:`eltype(IterType) <eltype>`                  ``Any``                  The type the items returned by :func:`next`
 :func:`length(iter) <length>`                      (*undefined*)            The number of items, if known
-:func:`iteratorsize(IterType) <iteratorsize>`      ``HasLength()``
-:func:`iteratoreltype(IterType) <iteratoreltype>`  ``HasEltype()``
-================================================== ======================== ===========================================
-
+:func:`size(iter, [dim...]) <size>`                (*undefined*)            The number of items in each dimension, if known
+================================================== ======================== =====================================================================================
+
+================================================================ ======================================================================
+Value returned by :func:`iteratorsize(IterType) <iteratorsize>`  Required Methods
+================================================================ ======================================================================
+`HasLength()`                                                    :func:`length(iter) <length>`
+`HasShape()`                                                     :func:`length(iter) <length>`  and :func:`size(iter, [dim...]) <size>`
+`IsInfinite()`                                                   (*none*)
+`SizeUnknown()`                                                  (*none*)
+================================================================ ======================================================================
+
+==================================================================== ==================================
+Value returned by :func:`iteratoreltype(IterType) <iteratoreltype>`  Required Methods
+==================================================================== ==================================
+`HasEltype()`                                                        :func:`eltype(IterType) <eltype>`
+`EltypeUnknown()`                                                    (*none*)
+==================================================================== ==================================
```

Comment:
you may need an extra blank line after the end of the table to avoid a sphinx error